src/solvers/prop/prop.h

```diff
@@ -94,7 +94,8 @@ class propt:public messaget, public prop_assignmentt
   // solving
   virtual const std::string solver_text()=0;
   enum class resultt { P_SATISFIABLE, P_UNSATISFIABLE, P_ERROR };
-  virtual resultt prop_solve()=0;
+  resultt operator()() { return prop_solve(); }
+  virtual resultt prop_solve()=0; // will eventually be protected
```

Comment:
Please mark as `/// \deprecated`

Reply:
Could I recommend this is a good opportunity to introduce `CPROVER_DEPRECATED` or similar, which will turn into the official `[[deprecated]]` attribute when compiled with c++14 or better?